src/Tutor.Core/BuildingBlocks/EventSourcing/EventSourcedEntity.cs

```diff
@@ -0,0 +1,21 @@
+﻿using System;
+
+namespace Tutor.Core.BuildingBlocks.EventSourcing
+{
+    public abstract class EventSourcedEntity : Entity
+    {
+        private EventSourcedAggregateRoot _root;
+
+        public void Initialize(EventSourcedAggregateRoot root)
+        {
+            _root = root ?? throw new ArgumentNullException(nameof(root));
+        }
```

Comment:
Since the EventSourcedEntity needs a reference to the EventSourcedAggregateRoot in order to work, I added an Initialize method to EventSourcedAggregateRoot which can then initialize all entities as needed. I wasn't quite sure where the best place to invoke the Initialize method is though, for now I placed it in the services before methods are called on KnowledgeComponentMastery. 

Also, since EventSourcedAggregate and EventSourcedEntity expect pretty specific behavior from inheriting classes, should I add some documentation comments explaining that? Most of it is pretty standard for event sourcing, but the addition of entities complicates things a bit so it might be good to have some extra clarity.

Reply:
Initialize moved to repositories in 5cc0ee555d6397b2c2e3a4add432510cd77a4983. Documentation comments added in 4140ef78c68f963478ce26a14c694ec78fa88160.